File: src/iphone_app/game_logic.swift
import Foundation

struct GameState {
    var currentContestant: Contestant
    var currentHost: Host
    var selectedCases: [Case]
    var remainingCases: [Case]
}

struct GameLogic {
    static var gameState: GameState?

    static func startGame() {
        gameState = GameState(currentContestant: Contestant(), currentHost: Host(), selectedCases: [], remainingCases: Case.generateCases())
        Interface.updateScore()
        Interface.updateCases()
        Interface.updateDialogue()
        Sound.playSound(name: "game-start")
    }

    static func selectCase(caseNumber: Int) {
        guard let index = gameState?.remainingCases.firstIndex(where: { $0.caseNumber == caseNumber }) else { return }
        let selectedCase = gameState?.remainingCases.remove(at: index)
        gameState?.selectedCases.append(selectedCase!)
        Interface.updateCases()
        Sound.playSound(name: "case-selected")
    }

    static func revealCase() {
        guard let selectedCase = gameState?.selectedCases.last else { return }
        gameState?.currentContestant.score += selectedCase.caseValue
        Interface.updateScore()
        Sound.playSound(name: "case-revealed")
    }
}

File: src/web_app/game_logic.js
var gameState = {
  currentContestant: null,
  currentHost: null,
  selectedCases: [],
  remainingCases: []
};

var startGame = function(contestantName) {
  gameState.currentContestant = new Contestant(contestantName);
  gameState.currentHost = new Host("Howie Mandel");
  gameState.selectedCases = [];
  gameState.remainingCases = generateCases();
  updateInterface();
};

var selectCase = function(caseNumber) {
  var selectedCase = gameState.remainingCases.find(function(c) {
    return c.number === caseNumber;
  });
  if (selectedCase) {
    gameState.selectedCases.push(selectedCase);
    gameState.remainingCases = gameState.remainingCases.filter(function(c) {
      return c.number !== caseNumber;
    });
    updateInterface();
  }
};

var revealCase = function() {
  var selectedCase = gameState.selectedCases.pop();
  if (selectedCase) {
    gameState.currentContestant.score += selectedCase.value;
    updateInterface();
  }
};

var generateCases = function() {
  var cases = [];
  for (var i = 1; i <= 26; i++) {
    cases.push(new Case(i, Math.floor(Math.random() * 1000000)));
  }
  return cases;
};

var updateInterface = function() {
  updateScore(gameState.currentContestant.score);
  updateCases(gameState.remainingCases);
  updateDialogue(gameState.currentHost.dialogue);
};